frontend/src/metabase/ui/components/inputs/TextareaBlurChange/TextareaBlurChange.tsx

```diff
@@ -60,6 +62,21 @@ export function TextareaBlurChange<T extends TextareaProps = TextareaProps>({
     [normalize, onBlur, onBlurChange, value],
   );
 
+  const handleKeyDown = useCallback(
+    (event: KeyboardEvent<HTMLTextAreaElement>) => {
+      if (event.key === "Escape") {
+        setInternalValue(value);
+
+        // setTimeout to allow internalValue to update before dispatching blur event
+        // so that blur event's target will have proper value
+        setTimeout(() => {
+          ref.current?.blur();
+        }, 0);
```

Comment:
(bis) Could manually setting the `ref.current.value` before calling `blur()` work too?
It'd be great to avoid the `setTimeout`.